system/Commands/Utilities/Routes/AutoRouterImproved/AutoRouteCollector.php

```diff
@@ -65,7 +65,7 @@ public function get(): array
 
         foreach ($finder->find() as $class) {
             // Exclude controllers in Defined Routes.
-            if (in_array($class, $this->protectedControllers, true)) {
+            if (in_array('\\' . $class, $this->protectedControllers, true)) {
```

Comment:
Would it make more sense to compare these as actual classes? Then we could rely more on autoloading and static analysis.

Reply:
What do you mean *to compare these as actual classes*?

Reply:
Something like `if (is_a($class, $protectedController, true)) ...`.

Reply:
> is_a — Checks if the object is of this class or has this class as one of its parents
https://www.php.net/manual/en/function.is-a.php

`or has this class as one of its parents` is not what we need.
It seems a good function for this does not exists.

Reply:
I assumed that would be a perk - do we ever want to allow extensions of a controller that is protected? If we only care about the class itself then `is_a()` isn't appropriate.

Reply:
When you have a controller A, and use it in a route definition.
You create `controller B extend A`, then do you expect to prohibit access to B via auto routing?

At least I did not expect that. 

It seems there is no good way to compare these as actual classes. 
So the current implementation in this PR seems to be okay.